### PR TITLE
Define Spec for Universe elements

### PR DESCRIPTION
**Pull request by BenWhitehead**
To view rendered doc: https://github.com/mesosphere/universe/blob/spec-doc/docs/spec.md

Layout feedback welcome, I'm not sure about the headers being used right now.

/cc @air @ConnorDoyle @jsancio

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+Schema: [config-schema.json](../repo/meta/schema/config-schema.json)
+
+### MUST namespace all properties
+
+*A single `options.json` should be able to be used for an entire DCOS Cluster*
+
+### MUST use hierarchical definition format
+
+1. MUST set `additionalProperties` to `false` for all nested objects
```
Comment (jsancio) [inline docs/spec.md]: Not all nested objects. I think we want this only on the package specific JSON object. For example I don't think we want it to be `false` for the Mesos' nested object.

**Comment by jsancio** [comment]
Very nice. Looks good to me.

**Comment by jsancio** [comment]
@BenWhitehead Should I merge this? Is this good to go?

**Comment by BenWhitehead** [comment]
Nope, not ready to merge yet. This can hold off for a bit.

**Comment by jsancio** [comment]
Sounds good. I added WIP label.

**Pull request closed without merging**